base/promotion.jl

```diff
@@ -403,20 +407,34 @@ where usually `^ == Base.^` unless `^` has been defined in the calling
 namespace.) If `y` is a negative integer literal, then `Base.literal_pow`
 transforms the operation to `inv(x)^-y` by default, where `-y` is positive.
 
+See also [`exp2`](@ref), [`<<`](@ref).
+
 # Examples
 ```jldoctest
 julia> 3^5
 243
 
-julia> A = [1 2; 3 4]
-2×2 Matrix{Int64}:
- 1  2
- 3  4
+julia> 3^-1  # uses Base.literal_pow
+0.3333333333333333
+
+julia> p = -1;
+
+julia> 3^p
+ERROR: DomainError with -1:
+Cannot raise an integer x to a negative power -1.
+[...]
+
+julia> 3.0^p
+0.3333333333333333
+
+julia> 10^19 > 0  # integer overflow
+false
+
+julia> big(10)^19 == 1e19
+true
 
-julia> A^3
-2×2 Matrix{Int64}:
- 37   54
- 81  118
```

Comment:
The method for matrix powers has its own docstring, with examples.

Reply:
Good catch! I dislike it when docstrings get too long with redundant examples across many methods. I often just want the first sentence or paragraph and the first thing I see at the REPL is a long list of examples for some esoteric method defined in a package.

Reply:
Am certainly guilty of extending quite a few... It's awkward because well-chosen examples, graded from simple to complicated, are actually very helpful. But scrolling past ones you don't care about is a pain. We have `??` for "extended help" but this doesn't seem like a great solution.